### PR TITLE
[reggen] - Update regwen support

### PR DESCRIPTION
**Pull request by tjaychen**
- Allow regwen to be a fully hw managed register

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
I'm not really sure if this will break any existing tests... especially since the hw managed regwens cannot be written to by software...

File: util/reggen/validate.py [hunk 1]
```diff
@@ -1296,25 +1300,28 @@ def check_wen_regs(regs):
     # 0 - name
     # 1 - reset value
     # 2 - sw access
+    # 3 - sw access
```
Comment (sjgitty) [inline util/reggen/validate.py:1303]: typo? hw access

File: util/reggen/validate.py [hunk 2]
```diff
@@ -1296,25 +1300,28 @@ def check_wen_regs(regs):
...
                 if 'name' in reg and 'genresval' in reg]
     mreg_list = [
         reg['multireg'] for reg in regs['registers'] if 'multireg' in reg
     ]
     field_list = [((reg['name'] + "_" + field['name']).lower(),
-                   field['genresval'], field['swaccess']) for mreg in mreg_list
-                  for reg in mreg['genregs'] for field in reg['fields']]
+                   field['genresval'], field['swaccess'], field['hwaccess'])
+                  for mreg in mreg_list for reg in mreg['genregs']
+                  for field in reg['fields']]
```
Comment (sjgitty) [inline util/reggen/validate.py:1318]: why does this indentation or lack thereof look strange? it appears you've run this through yapf

**Comment by eunchan** [comment]
LGTM other than the typo as Scott mentioned, also probably the last `if` part ..
`if not ( (xxx) or (yyy and zzz)):`

File: util/reggen/validate.py (hunk 1 above)
Comment (tjaychen) [reply util/reggen/validate.py:1303]: thanks will fix.

**Comment by tjaychen** [comment]
> LGTM other than the typo as Scott mentioned, also probably the last `if` part ..
> `if not ( (xxx) or (yyy and zzz)):`

ah yes i just ran `lintpy.py`. Should I have done something else instead?

**Comment by tjaychen** [comment]
wait sorry, i didn't catch what i should change with the last IF part.

**Comment by tjaychen** [comment]
i updated the long IF check a bit.  Still not too sure what to do about lint checks though...

**Comment by eunchan** [comment]
No, that is fine. I also fall into the same situation frequently when I 
use yapf. yapf breaks readibility sometimes in my opinion.

File: util/reggen/validate.py [hunk 3]
```diff
@@ -1328,10 +1335,24 @@ def check_wen_regs(regs):
...
+        hw_regwen = 0
+
+        if reg_list[idx][tuple_swaccess] in ["rw0c", "rw1c"]:
+            sw_regwen += 1
+
+        if reg_list[idx][tuple_swaccess] == "ro" and reg_list[idx][
+                tuple_hwaccess] == "hwo":
+            hw_regwen += 1
+
+        if (sw_regwen + hw_regwen) == 0:
```
Comment (eunchan) [inline util/reggen/validate.py:1350]: Nice! :) Thanks for revising it.

**Pull request merged**